src/util/Util.js

```diff
@@ -78,8 +78,8 @@ class Util {
   static parseEmoji(text) {
     if (text.includes('%')) text = decodeURIComponent(text);
     if (text.includes(':')) {
-      const [name, id] = text.split(':');
-      return { name, id };
+      const [, animated, name, id] = text.match(/<?(a)?:([^:]{2,32}):(\d{17,19})>?/);
```

Comment:
Only letters numbers and underscore is allowed in custom emoji names. So `[^:]` should be substituted with `\w`.